Add typescript templates

zackdotcomputer opened the pull request:
Adds two templates showing how to use this component in typescript with either the vanilla tsc or with webpack as your compiler/bundler.

This was done to explore whether it was even possible after reading #12

eahefnawy commented on templates/express-starter-typescript-tsc/serverless.yml:
```diff
@@ -0,0 +1,9 @@
+app: express-starter
+component: express
+name: express-starter
```
could you change that to `express-starter-typescript-tsc`?

eahefnawy commented on templates/express-starter-typescript-webpack/serverless.yml:
```diff
@@ -0,0 +1,9 @@
+app: express-starter
+component: express
+name: express-starter
```
could you change that to `express-starter-typescript-webpack`?

eahefnawy reviewed (changes requested):
Thanks for the PR! DId you face any issues getting it to work with TypeScript?

zackdotcomputer commented:
@eahefnawy Not at all - was able to pipe the existing templates through tsc or webpack just fine and, once I added the typing, everything seemed to work ok.

Made the changes you requested and fixed the lint/prettier issues that the GH Action turned up

eahefnawy reviewed (approved):
Thank you 🙏
